delete-tags.js

```diff
@@ -0,0 +1,49 @@
+module.exports = async ({ github, context, core }) => {
+  const { IMAGE_NAME, ORGANIZATION } = process.env
+
+  let tagName
+
+  if (context.payload.ref_type === 'branch') {
+    tagName = `branch-${context.payload.ref}`
+  } else {
+    tagName = context.payload.ref.match(/^v(.*)$/)[1]
+  }
+
+  let didDelete = false
+
+  for await (const response of github.paginate.iterator(
+    github.rest.packages.getAllPackageVersionsForPackageOwnedByOrg, {
+      package_type: 'container',
+      package_name: IMAGE_NAME,
+      org: ORGANIZATION
+    })) {
+    for (const version of response.data) {
+      const tags = version.metadata?.container?.tags
+      if (tags?.includes(tagName)) {
+        core.notice(`Package version ${version.html_url} matches tag ${tagName} and will be deleted`)
+
+        const otherTags = tags.filter((tag) => tag !== tagName)
```

Comment:
**suggestion (non-blocking):**

Can't this check `tags.length > 1` rather than filtering and checking `otherTags.length`? We already know `tags` has `tagName`, so presumably any other items would be not-`tagName`.

Reply:
Yeah, that probably makes more sense - I'm not really sure why I did it this way. I'm inclined to leave it since I tested and confirmed that `tags.filter` works, unless you want to upgrade this to a blocker.